File: 1196/history.c
#include <stdio.h>

int prof_nums[15001];
int n, m;
int key;

int binary_search() 
{
	int l = 0;
	int h = n - 1;
	int mid;

	while (l <= h) {
		mid = (l + h) / 2;

		if (prof_nums[mid] == key)
			return mid;
		else if (prof_nums[mid] < key) 
			l = mid + 1;
		else
			h = mid - 1;
	}

	return -1;
}

int main(int argc, char *argv[]) 
{
	int i, t, count = 0;
	
	scanf("%d", &n);

	for(i=0; i<n; i++)
		scanf("%d", &prof_nums[i]);

	scanf("%d", &m);

	for(i=0; i<m; i++) {
		scanf("%d", &key);

		t = binary_search();

		if (t != -1)
			count++;
	}

	printf("%d\n", count);


	return 0;
}

File: 1196/history.py
def binary_search(A, n, k):
	l = 0
	h = n - 1

	while l <= h:
		mid = (l + h) / 2

		if k == A[mid]:
			return mid
		elif k < A[mid]:
			h = mid - 1
		else:
			l = mid + 1

	return -1


n = int(raw_input())

x = []

for i in range(n):
	t = int(raw_input())	
	x.append(t)

# prof_nums = set(x)

m = int(raw_input())
count = 0

for i in range(m):
	t = int(raw_input())

	# if t in prof_nums:
	y = binary_search(x, n, t)
	# print t, y
	if  y != -1:
		count = count + 1

print count

